### PR TITLE
NDS-85 Improve print function

### PR DESCRIPTION
**Pull request by greypiglet**
https://nicedigital.atlassian.net/browse/NDS-85

### CODE REVIEW

File: components/nds-core/scss/helpers/_helpers-print.scss [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+@mixin nds-no-appended-print-link {
+  &:after {
+    content: none !important;
+  }
+}
+
+///Helper for including print styling
+/// @output CSS to wrap a media query with content
+/// @since 1.0.0
+@mixin nds-print {
```
Comment (ediblecode) [inline components/nds-core/scss/helpers/_helpers-print.scss]: Not convinced this mixin is worth having over and above native @media print?

File: components/nds-core/scss/helpers/_helpers-print.scss [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    display: none !important;
+  }
+}
+
+///Only show the element when printing (not for other media)
+/// @output the CSS to hide on all media except print
+/// @since 1.0.0
+@mixin nds-show-print {
+  display: none !important;
+  @media print {
```
Comment (ediblecode) [inline components/nds-core/scss/helpers/_helpers-print.scss]: If we're going to keep @mixin nds-print then we should use it here

File: components/nds-core/scss/helpers/_helpers-print.scss [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    display: block !important;
+  }
+}
+
+///Remove an appended link URL
+/// @output removes content attribute
+/// @since 1.0.0
+@mixin nds-no-appended-print-link {
+  &:after {
```
Comment (ediblecode) [inline components/nds-core/scss/helpers/_helpers-print.scss]: This is missing a print media query

File: components/nds-core/scss/helpers/_helpers-print.scss (hunk 2 above)
Comment (ediblecode) [reply components/nds-core/scss/helpers/_helpers-print.scss]: Also, we should then be consistent with all the other print media queries e.g. https://github.com/nice-digital/nice-design-system/blob/v1/components/design-system/scss/typography/_typography-quotations.scss#L29-L31

File: components/nds-core/scss/helpers/_helpers-print.scss [hunk 4]
```diff
@@ -0,0 +1,42 @@
+////
+/// @group helpers
+////
+
+/// Styles to hide and show under the print media query
+
+/// Hide the element when printing
+/// @output the CSS to hide on the print stylesheet
+/// @since 1.0.0
+@mixin nds-hide-print {
```
Comment (ediblecode) [inline components/nds-core/scss/helpers/_helpers-print.scss]: Can you apply this to the other places where we're already hiding for print e.g. https://github.com/nice-digital/nice-design-system/blob/v1/components/nds-stacked-nav/scss/stacked-nav.scss#L87-L89 (I think we do it for CTA in page header and there are probably other use cases).

File: components/design-system/scss/helpers/_helpers-print.scss [hunk 5]
```diff
@@ -38,24 +26,30 @@
...
+    h5 a,
+    h6 a {
+      @include nds-no-appended-print-link;
+    }
+
     a[href]:after {
       content: ' (' attr(href) ')';
     }
 
+    a.show-link[href]:after {
```
Comment (ediblecode) [inline components/design-system/scss/helpers/_helpers-print.scss]: Can you combine this with the rule above? ...but also, do we need it?

Comment (ediblecode) [reply components/design-system/scss/helpers/_helpers-print.scss]: Or maybe it should be called `append-print-link` or something?

Comment (greypiglet) [reply components/design-system/scss/helpers/_helpers-print.scss]: I personally can't imagein this being used - will remove

**Comment by greypiglet** [comment]
@ediblecode hoping this is ready as well

**Pull request merged**